modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaEntityBrowserWidgetTest.php

```diff
@@ -0,0 +1,253 @@
+    $this->assertOrderInPage(['Image 1', 'Image 2']);
+
+    // Fill in the other fields and save the node.
+    $this->getSession()->getPage()->fillField('featured_media_field[0][caption]', 'Image 1 caption');
+    $this->getSession()->getPage()->fillField('featured_media_field[1][caption]', 'Image 2 caption');
+    $this->getSession()->getPage()->fillField('Title', 'Test entity browser widget');
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert that the values were saved.
+    $this->assertSession()->pageTextContains('Featured media field');
+    $this->assertSession()->pageTextContains('Image 1');
+    $this->assertSession()->pageTextContains('Image 2');
+    $this->assertSession()->pageTextContains('Image 1 caption');
+    $this->assertSession()->pageTextContains('Image 2 caption');
+    // Assert items order is the same as it was in the form.
+    $this->assertOrderInPage(['Image 1', 'Image 2']);
+
+    // Edit the node to reorder field items.
+    $this->drupalGet('node/1/edit');
+    file_put_contents('test.html', $this->getSession()->getPage()->getOuterHtml());
+    $dragged = $this->getSession()->getPage()->find('css', 'table#featured-media-field-values > tbody > tr:nth-child(1) a.tabledrag-handle');
+    $target = $this->getSession()->getPage()->find('css', 'table#featured-media-field-values > tbody > tr:nth-child(2) a.tabledrag-handle');
+    $dragged->dragTo($target);
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Check that 'Image 1' media item is placed after 'Image 2'.
+    $this->assertOrderInPage(['Image 2', 'Image 1']);
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert items order was saved.
+    $this->assertOrderInPage(['Image 2', 'Image 1']);
+
+    // Edit the node to remove the first item.
+    $this->drupalGet('node/1/edit');
+    $this->getSession()->getPage()->pressButton('Remove');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    // Assert the image was removed from the field.
+    $this->assertSession()->pageTextNotContains('Image 2');
+    $this->assertSession()->buttonExists('Select images');
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert that the update was saved.
+    $this->assertSession()->pageTextContains('Featured media field');
+    $this->assertSession()->pageTextContains('Image 1');
+    $this->assertSession()->pageTextContains('Image 1 caption');
+  }
+
+  /**
+   * Asserts items are placed in markup in the given order.
+   */
+  protected function assertOrderInPage(array $items) {
+    $session = $this->getSession();
+    $text = $session->getPage()->getHtml();
+    $strings = [];
+    foreach ($items as $item) {
+      if (($pos = strpos($text, $item)) === FALSE) {
+        throw new ExpectationException("Cannot find '$item' in the page", $session->getDriver());
+      }
+      $strings[$pos] = $item;
+    }
+    ksort($strings);
+    $ordered = implode(', ', array_map(function ($item) {
```

Comment:
What? I think you meant to order the actual found strings and not the passed ones no? I don't think this is even needed. You can assert the same and implode the actual found strings no?

Reply:
$ordered just makes one string from the given items to be returned when the order is not the same. The message returned is indeed wrong.